contracts/payment/MCPayment.sol

```diff
@@ -116,12 +147,99 @@ contract MCPayment is Ownable2StepUpgradeable, EIP712Upgradeable {
         $.isPaid[paymentId] = true;
     }
 
+    function payERC20(
+        Iden3PaymentRailsERC20RequestV1 memory paymentData,
+        bytes memory signature
+    ) external {
+        verifyIden3PaymentRailsERC20RequestV1Signature(paymentData, signature);
+        bytes32 paymentId = keccak256(abi.encode(paymentData.recipient, paymentData.nonce));
+        MCPaymentStorage storage $ = _getMCPaymentStorage();
+        if ($.isPaid[paymentId]) {
+            revert PaymentError(
+                paymentData.recipient,
+                paymentData.nonce,
+                "MCPayment: payment already paid"
+            );
+        }
+        IERC20 token = IERC20(paymentData.tokenAddress);
+        if (token.transferFrom(msg.sender, address(this), paymentData.amount)) {
+            uint256 ownerPart = (paymentData.amount * $.ownerPercentage) / 100;
+            uint256 issuerPart = paymentData.amount - ownerPart;
+            token.transfer(paymentData.recipient, issuerPart);
+            emit Payment(paymentData.recipient, paymentData.nonce);
+            $.isPaid[paymentId] = true;
+        } else {
+            revert PaymentError(
+                paymentData.recipient,
+                paymentData.nonce,
+                "MCPayment: ERC-20 transfer failed"
+            );
+        }
+    }
+
+    function payERC20Permit(
+        bytes memory permitSignature,
+        Iden3PaymentRailsERC20RequestV1 memory paymentData,
+        bytes memory signature
+    ) external {
+        verifyIden3PaymentRailsERC20RequestV1Signature(paymentData, signature);
+        bytes32 paymentId = keccak256(abi.encode(paymentData.recipient, paymentData.nonce));
+        MCPaymentStorage storage $ = _getMCPaymentStorage();
+        if ($.isPaid[paymentId]) {
+            revert PaymentError(
+                paymentData.recipient,
+                paymentData.nonce,
+                "MCPayment: payment already paid"
+            );
+        }
```

Comment:
This code is repeated in both `payERC20` and `payERC20Permit`. Maybe we could create a function `checkERC20Payment(paymentData, signature)` or something similar with the code.